src/assets/Generator.Shared/LowLevelBinaryDataOperation.cs

```diff
@@ -12,47 +12,14 @@
 
 namespace Azure.Core
 {
-    /// <summary>
-    /// The common operation type used by all low level client methods. The result of the operation is the `Response.Content` property of the final state.
-    /// </summary>
-    internal class LowLevelBinaryDataOperation : Operation<BinaryData>, IOperationSource<BinaryData>
+    internal class LowLevelBinaryDataOperation : FuncOperation<BinaryData>
```

Comment:
What is the other type derived from `FuncOperation`?

Reply:
Nothing else derives from it, but we code-gen instances of `FuncOperation<BinaryData>` to return in cases where the result of the LRO is pageable (https://github.com/Azure/autorest.csharp/pull/1514/files/57772ced9de4fff85a47bf465a96adb3e61d2e01#diff-dd69b9ac81b22be268f126e383ffffde84f2520679523a6691c808008c1fd3a6R213)

Reply:
If you would rather `LowLevelBinaryDataOperation` not inherit from `FuncOperations` (and have me seal both types) I'm happy to do that.  I extracted the guts of `LowLevelBinaryDataOperation` into `FuncOperation<BinaryData>` so we had an easy way to return an operation which returns a pageable without having to go codegen a specific type per operation (which we wouldn't want to expose).

Reply:
@AlexanderSher Thanks for explaining your concern to me earlier today.  I pushed https://github.com/ellismg/autorest.csharp/pull/1514/commits/2c6b0d9f29c976f6ab0cb0d4fae8b3dc08658389 to the PR which removes the `LowLevelBinaryDataOperation` type in favor of just using `LowLevelFuncOperation<T>` everywhere.  I think this gets at the issue you raised, but please let me know if you'd like something else done.

Thanks again for all your help improving the code here and the very in depth review!